### PR TITLE
fix: change headless service to gRPC and expose 9094 TCP

### PR DESCRIPTION
**Pull request by locmai**
## What this PR does

Follow-up the work of @AlexandreRoux from #435 and related to #420 to run the helm-docs for the README.md

> This PR corrects the alertmanager headless service with exposing gRPC port instead of HTTP.

I tested the Helm chart and I can't reproduce the EOF issue.

### CODE REVIEW

File: templates/alertmanager/alertmanager-svc-headless.yaml [hunk 1]
```diff
@@ -17,11 +16,18 @@ spec:
...
+      name: grpc
+      targetPort: grpc
+    - port: 9094
+      protocol: UDP
+      name: alert-clu-udp
+      targetPort: alert-clu-udp
+    - port: 9094
+      protocol: TCP
+      name: alert-clu-tcp
+      targetPort: alert-clu-tcp
```
Comment (nschad) [inline templates/alertmanager/alertmanager-svc-headless.yaml:30]: We should somehow notify the user that if the service already exists, that this will not work properly due to https://github.com/kubernetes/kubernetes/issues/39188

File: templates/alertmanager/alertmanager-svc-headless.yaml [hunk 2]
```diff
@@ -17,11 +16,18 @@ spec:
   clusterIP: None
   publishNotReadyAddresses: true
   ports:
-    - port: {{ .Values.config.server.http_listen_port }}
+    - port: {{ .Values.config.server.grpc_listen_port }}
       protocol: TCP
-      name: http-metrics
-      targetPort: http-metrics
```
Comment (nschad) [inline templates/alertmanager/alertmanager-svc-headless.yaml:23]: we should inform the user about this breaking chance in CHANGELOG.md

File: templates/alertmanager/alertmanager-svc-headless.yaml [hunk 3]
```diff
@@ -1,5 +1,4 @@
 {{- if .Values.alertmanager.enabled -}}
-{{- if .Values.alertmanager.statefulSet.enabled -}}
```
Comment (nschad) [inline templates/alertmanager/alertmanager-svc-headless.yaml:2]: Why was this removed?

Comment (locmai) [reply templates/alertmanager/alertmanager-svc-headless.yaml:2]: I think that was Alex's change respond to your previous comment about the deployment without the headless service

> Also currently if you set statefulSet to false (the default) and deploy cortex then you only get the non-headless service of alertmanager. Makes no sense.

So he reverted the removal of the dep.yaml and also had this in the next commit.

File: templates/alertmanager/alertmanager-svc-headless.yaml (hunk 2 above)
Comment (locmai) [reply templates/alertmanager/alertmanager-svc-headless.yaml:23]: I have updated the CHANGELOG.md to make this clearer. Or do you think we should add a better indication that this may break if users were using the `http-metrics` port for their own usage? Similar to the old PR we have here https://github.com/cortexproject/cortex-helm-chart/pull/169

And by breaking - do you mean we should make it to v3 (too much I think?) or a line in the CHANGELOG should be good enough?

File: templates/alertmanager/alertmanager-svc-headless.yaml (hunk 1 above)
Comment (locmai) [reply templates/alertmanager/alertmanager-svc-headless.yaml:30]: Changelog updated.

File: templates/alertmanager/alertmanager-svc-headless.yaml (hunk 2 above)
Comment (nschad) [reply templates/alertmanager/alertmanager-svc-headless.yaml:23]: Yep I think this should warrant a v3.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 ## master / unreleased
 
+* [CHANGE] Replace `http-metrics` port with `grpc` port #494
```
Comment (nschad) [inline CHANGELOG.md:5]:
```suggestion
* [CHANGE] Alertmanager: Replace `http-metrics` port with `grpc` port #494 
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,10 @@
 
 ## master / unreleased
 
+* [CHANGE] Replace `http-metrics` port with `grpc` port #494 
+* [CHANGE] Expose 9094 TCP and UDP for gossip cluster - #494
```
Comment (nschad) [inline CHANGELOG.md:6]:
```suggestion
* [CHANGE] Alertmanager: Expose 9094 TCP and UDP for gossip cluster - #494
```

**Review by nschad — APPROVED** [review]

File: templates/alertmanager/alertmanager-svc-headless.yaml (hunk 2 above)
Comment (kd7lxl) [reply templates/alertmanager/alertmanager-svc-headless.yaml:23]: I would just not touch the `http-metrics` port, so that there is no breaking change. Adding gRPC has nothing to do with the `http-metrics` port.

Comment (locmai) [reply templates/alertmanager/alertmanager-svc-headless.yaml:23]: That sounds good. Let me update that.

Comment (locmai) [reply templates/alertmanager/alertmanager-svc-headless.yaml]: Done!

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 ## master / unreleased
 
+* [CHANGE] Alertmanager: Add `grpc` port #494 
+* [CHANGE] Alertmanager: Expose 9094 TCP and UDP for gossip cluster #494
```
Comment (kd7lxl) [inline CHANGELOG.md:6]: These should be `ENHANCEMENT`.

**Review by kd7lxl — APPROVED** [review]
